--- uim-1.8.6/configure.ac~	2013-06-30 06:26:20.000000000 +0200
+++ uim-1.8.6/configure.ac	2015-04-12 14:21:41.827256716 +0200
@@ -898,12 +898,12 @@
         use_applet_gnome3="no"
       ;;
       yes|*)
-        PKG_CHECK_MODULES(APPLET_GNOME3, libpanelapplet-4.0, use_applet_gnome3="yes",use_applet_gnome3="no")
+        PKG_CHECK_MODULES(APPLET_GNOME3, libpanel-applet, use_applet_gnome3="yes",use_applet_gnome3="no")
       ;;
     esac
   ],
   [
-    PKG_CHECK_MODULES(APPLET_GNOME3, libpanelapplet-4.0, use_applet_gnome3="yes",use_applet_gnome3="no")
+    PKG_CHECK_MODULES(APPLET_GNOME3, libpanel-applet, use_applet_gnome3="yes",use_applet_gnome3="no")
   ])
 
 dnl ****************************
--- uim-1.8.6/gtk3/toolbar/applet-gnome3.c~	2013-06-30 06:26:09.000000000 +0200
+++ uim-1.8.6/gtk3/toolbar/applet-gnome3.c	2015-04-12 14:36:02.283917154 +0200
@@ -168,7 +168,7 @@
   gtk_action_group_set_translation_domain(action_group, GETTEXT_PACKAGE);
   gtk_action_group_add_actions(action_group, uim_menu_actions,
       G_N_ELEMENTS(uim_menu_actions), toolbar);
-  panel_applet_setup_menu(applet, uim_menu_xml, action_group);
+  panel_applet_setup_menu(applet, uim_menu_xml, action_group, "uim");
 #if LIBPANEL_APPLET_HAVE_SET_BACKGROUND_WIDGET
   panel_applet_set_background_widget(applet, GTK_WIDGET(applet));
 #endif
--- uim-1.8.6/gtk3/toolbar/Makefile.am~	2013-06-30 06:26:09.000000000 +0200
+++ uim-1.8.6/gtk3/toolbar/Makefile.am	2015-04-12 14:50:19.527244281 +0200
@@ -18,7 +18,7 @@
 uim_toolbar_applet_gnome3_SOURCES = applet-gnome3.c \
 			       ../../gtk2/toolbar/common-gtk.c
 
-appletdir = $(datadir)/gnome-panel/4.0/applets
+appletdir = $(datadir)/gnome-panel/5.0/applets
 applet_DATA = UimApplet.panel-applet
 applet_in_files = $(applet_DATA:=.in)
 applet_in_in_files = $(applet_in_files:=.in)
